src/YAMLConfiguration.cpp

```diff
@@ -224,8 +224,6 @@ bool libConfig::YAMLConfigParser::loadConfigString(const std::string &yamlstring
 template <typename T>
 bool libConfig::YAMLConfigParser::loadConfig(T &stream, std::map<std::string, libConfig::Configuration> &subConfigs)
 {
-    subConfigs.clear();
```

Comment:
could you please elaborate a bit about the need for this? E.g.: in which cases does it case you've observed which problems. Maybe with a brief (pseudo) code example.
I can only imaging the reason might be, that you try loading multiple files sequentially into the same map?

Reply:
Isn't it a very common scenario to load multiple files and merge there configurations? The trenhancer does this internally, that is why I assumed that merging task configurations in this way is the desired behaviour.

See here for the usage:
https://git.hb.dfki.de/rock-dfki/cnd-service-trenhancer/-/blob/rework/src/RuntimeModelExecuter.cpp#L351

In the current situation this leads to only the last configuration is loaded and all others being ignored.

On a site note: I actually do not think that a user of Bundles (in this case trenhancer) should manually merge configurations this way. Instead Bundle should do this internally and just provide the resulting configurations for a given Task. But such a functionality seems not available, so the way it is handled within trenhancer seems the most straight forward way at the moment.

Reply:
> Isn't it a very common scenario to load multiple files and merge their configurations?

It *was* no common scenario, since it did not work :), but it is supposed to work like that.
When bundles are used, the need to load multiple config files into the same object occurs, if multiple bundles are selected that provide config files for the same tasks. Is this also the case where you experienced the problem?

Don't get me wrong: I'm not asking, because I think the patch is wrong. I just want to fully understand the case you're testing.
The 2nd best possible outcome from my perspective is that you say that config file loading over multiple selected bundles now works properly.
The best possible outcome would be that there's a unit test case for this in https://github.com/rock-cpp/lib_config/tree/master/test :)

Reply:
Yes, it happens when I several bundles provide configurations for a task.

I will have a look into the unit test.